main/prod/python/binance.py

```diff
@@ -0,0 +1,387 @@
+
+        
+        return result
+
+
+    async def get_orderbook(self, pairs_to_filter: list[str], pair: str = None):
+        endpoint = "/fapi/v1/ticker/bookTicker"
+
+        binance_out = 1
+
+        if pair:
+            params = {
+                "symbol": pair,
+            }
+        else:
+            params = None
+
+        while binance_out:
+            try:
+                async with aiohttp.ClientSession() as session:
+                    exchange_info = await session.get(self.BASE_ENDPOINT + endpoint, params=params, ssl=False)
+                    result = await exchange_info.json()
+                binance_out = 0
+            except Exception as e:
+                print(e)
+                time.sleep(1)
+        
+        # filtering results for only pairs in intersections between the two exchanges:
+        filtered_result = [pair for pair in result if pair["symbol"] in pairs_to_filter]
+
+        # ordering results based on symbol name:
+        ordered_result = sorted(filtered_result, key=lambda i: i["symbol"])
+
+        return ordered_result
+        # return [float(result["bidPrice"]), float(result["askPrice"])]
+
+
+
+        # Possible intervals for klines
+        # 1m
+        # 3m
+        # 5m
+        # 15m
+        # 30m
+        # 1h
+        # 2h
+        # 4h
+        # 6h
+        # 8h
+        # 12h
+        # 1d
+        # 3d
+        # 1w
+        # 1M
+    def get_kline(self, pair: str, interval: str, startTime: str, endTime = round(time.time() * 1000), limit: int = 1500):
+        # current time code = round(time.time() * 1000)
+        endpoint = "/fapi/v1/klines"
+        params = {
+            "symbol": pair,
+            "interval": interval,
+            # "startTime": management.date_to_ms(startTime),
+            'startTime': startTime,
+            "endTime": endTime,
+            "limit": limit
+        }
+
+        try:
+            kline = requests.get(
+                self.BASE_ENDPOINT + endpoint, params=params).json()
+            binance_out = 0
+        except Exception as e:
+            print(e)
+        
+        df = pd.DataFrame(kline, columns=['Date', 'Open', 'High', 'Low', 'Close', 'Volume', 'Close_time', 'Quote_asset_volume', 'Number_of_trades', 'Tbbav', 'Tbqav', 'Ignore'])
+        df['Date'] = pd.to_datetime(df['Date'], unit='ms')
+        df.drop(['Close_time','Quote_asset_volume', 'Number_of_trades', 'Tbbav', 'Tbqav', 'Ignore'], axis=1, inplace=True)
+        df['Open'] = pd.to_numeric(df['Open'], errors='coerce')
+        df['High'] = pd.to_numeric(df['High'], errors='coerce')
+        df['Low'] = pd.to_numeric(df['Low'], errors='coerce')
+        df['Close'] = pd.to_numeric(df['Close'], errors='coerce')
+        df['Volume'] = pd.to_numeric(df['Volume'], errors='coerce')
+        df.set_index('Date', inplace=True)
+        return df
+
+
+    def time_intervals_to_seconds(self, interval: str) -> int:
```

Comment:
Me parece que esse método não está sendo mais utilizado. Se for verdade, sugiro removê-lo!